src/header.jl

```diff
@@ -362,7 +363,7 @@ function show(io::IO, hdr::FITSHeader)
                     @printf io " / %s" hdr.comments[i]
                 end
             end
-            print(io, "\n")
+            i != n && print(io, "\n")
```

Comment:
Looks like there will still be a trailing newline if the last keyword is a HISTORY or COMMENT (L344 and L346).